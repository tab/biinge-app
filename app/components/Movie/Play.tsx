import React, { ComponentType, useState } from "react"
import { compose } from "@reduxjs/toolkit"
import { Modal, Pressable } from "react-native"
import Animated, { FadeIn } from "react-native-reanimated"

import { useMovieVideos } from "hocs"
import Player from "components/ui/Player"
import Icon from "components/ui/Icon"
import { MovieVideo } from "types"
import colors from "styles/colors"
import { playStyles } from "styles"

type Props = {
  items: MovieVideo[]
}

const PlayComponent = ({ items }: Props) => {
  const [visible, setVisible] = useState(false)
  const trailer = items[0]

  const handlePlayer = () => {
    setVisible(!visible)
  }

  return (
    <>
      {trailer && (
        <Animated.View entering={FadeIn} style={playStyles.root}>
          <Pressable style={playStyles.button} onPress={handlePlayer}>
            <Icon name="play" color={colors.white} size={24} />
          </Pressable>
          <Modal transparent animationType="none" visible={visible}>
            <Player videoId={trailer.key} onClose={handlePlayer} />
          </Modal>
        </Animated.View>
      )}
    </>
  )
}

export default compose<ComponentType>(useMovieVideos)(PlayComponent)
