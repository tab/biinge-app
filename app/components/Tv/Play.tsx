import React, { ComponentType, useState } from "react"
import { compose } from "@reduxjs/toolkit"
import { Modal, Pressable } from "react-native"
import Animated, { FadeIn } from "react-native-reanimated"

import { useTvVideos } from "hocs"
import Player from "components/ui/Player"
import Icon from "components/ui/Icon"
import { TvVideo } from "types"
import colors from "styles/colors"
import { playStyles } from "styles"

type Props = {
  items: TvVideo[]
}

const PlayComponent = ({ items }: Props) => {
  const [visible, setVisible] = useState(false)
  const trailer = items[0]

  const handlePlayer = () => {
    setVisible(!visible)
  }

  return (
    <>
      {trailer && (
        <Animated.View entering={FadeIn} style={playStyles.root}>
          <Pressable style={playStyles.button} onPress={handlePlayer}>
            <Icon name="play" color={colors.white} size={24} />
          </Pressable>
          <Modal transparent animationType="none" visible={visible}>
            <Player videoId={trailer.key} onClose={handlePlayer} />
          </Modal>
        </Animated.View>
      )}
    </>
  )
}

export default compose<ComponentType>(useTvVideos)(PlayComponent)
