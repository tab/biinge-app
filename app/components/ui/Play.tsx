import React, { useState } from "react"
import { Modal, Pressable } from "react-native"
import Animated, { ZoomIn } from "react-native-reanimated"

import Player from "components/ui/Player"
import Icon from "components/ui/Icon"
import { Video } from "types"
import colors from "styles/colors"
import { playStyles } from "styles"

type Props = {
  items: Video[]
}

const PlayComponent = ({ items }: Props) => {
  const [visible, setVisible] = useState(false)
  const trailer = items[0]

  const handlePlayer = () => {
    setVisible(!visible)
  }

  return (
    <>
      {trailer && (
        <Animated.View entering={ZoomIn.delay(500)} style={playStyles.root}>
          <Pressable style={playStyles.button} onPress={handlePlayer}>
            <Icon name="play" color={colors.white} size={24} />
          </Pressable>
          <Modal transparent animationType="none" visible={visible}>
            <Player videoId={trailer.key} onClose={handlePlayer} />
          </Modal>
        </Animated.View>
      )}
    </>
  )
}

export default PlayComponent
