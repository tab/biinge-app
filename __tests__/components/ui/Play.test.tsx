import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"
import { it } from "@jest/globals"

import Play from "components/ui/Play"

describe("Play", () => {
  const videos = [
    {
      id: "123",
      tmdbId: 123,
      name: "Test Trailer",
      key: "abc123",
      publishedAt: "2023-01-01T12:00:00Z",
    },
  ]

  it("does not render when there are no videos", () => {
    const tree = render(<Play items={[]} />)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly with video items", () => {
    const tree = render(<Play items={videos} />)
    expect(tree).toMatchSnapshot()
  })
})
