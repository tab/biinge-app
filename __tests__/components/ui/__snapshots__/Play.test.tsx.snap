// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Play does not render when there are no videos 1`] = `null`;

exports[`Play renders correctly with video items 1`] = `
<View
  entering="ZoomIn"
  style={
    {
      "height": 40,
      "position": "absolute",
      "right": 65,
      "top": 767.5,
      "width": 40,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "borderRadius": 50,
        "height": "100%",
        "justifyContent": "center",
        "paddingLeft": 3,
        "width": "100%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#ffffff",
            "fontSize": 24,
          },
          undefined,
          {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î·‡
    </Text>
  </View>
</View>
`;
